### PR TITLE
Service warmups

### PR DESCRIPTION
**Pull request by jonsequitur**
This introduces a new interface, `IPerformServiceWarmup`, that can be used to warm up a service before any tests are executed. It is registered during test target setup by calling `UseServiceWarmup`.

### CODE REVIEW

File: Peaky.Tests/ServiceWarmupTests.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                                             .GetAsync("http://blammo.com/tests/production/widgetapi/Slow_test");
+
+            response.ShouldSucceed(HttpStatusCode.OK);
+        }
+
+        [Fact]
+        public async Task Warm_up_status_is_cached()
+        {
+            await peakyService.CreateHttpClient()
+                              .GetAsync("http://blammo.com/tests/production/widgetapi/Fast_test");
```
Comment (TyOverby) [inline Peaky.Tests/ServiceWarmupTests.cs:51]: Why is this URL used?  It doesn't seem to be an actual testing site?

File: Peaky/ServiceWarmupTracker.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+        private bool warmedUp;
+
+        public ServiceWarmupTracker(IPerformServiceWarmup warmup)
+        {
+            this.warmup = warmup ?? throw new ArgumentNullException(nameof(warmup));
+
+            timer = new Timer(15000);
+
+            timer.Elapsed += (sender, args) => warmedUp = false;
```
Comment (rchande) [inline Peaky/ServiceWarmupTracker.cs:24]: Why?

**Review by rchande — APPROVED** [review]

File: Peaky/ServiceWarmupTracker.cs (hunk 2 above)
Comment (jonsequitur) [reply Peaky/ServiceWarmupTracker.cs:24]: The target service will periodically restart and need warmups again.

File: Peaky.Tests/ServiceWarmupTests.cs (hunk 1 above)
Comment (jonsequitur) [reply Peaky.Tests/ServiceWarmupTests.cs:51]: The tests intercept the requests (take a look at the `PeakyService` class), so no outbound calls are actually made. The domain is not significant.

**Pull request merged**